transformer/transformers/ampruntimejs.go

```diff
@@ -0,0 +1,37 @@
+package transformers
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/ampproject/amppackager/transformer/internal/amphtml"
+	"github.com/ampproject/amppackager/transformer/internal/htmlnode"
+	"golang.org/x/net/html"
+	"golang.org/x/net/html/atom"
+)
+
+// AMPRuntimeJS rewrites the value of src in script nodes, where applicable.
+// If the value is of the form "*.js", replace it with "*.sxg.js".
+func AMPRuntimeJS(e *Context) error {
+	for n := e.DOM.RootNode; n != nil; n = htmlnode.Next(n) {
+		if n.Type != html.ElementNode {
+			continue
+		}
+		if n.DataAtom == atom.Script {
+			src, ok := htmlnode.FindAttribute(n, "", "src")
+			if ok && strings.HasPrefix(src.Val, amphtml.AMPCacheRootURL) {
+				u, _ := url.Parse(src.Val)
+				query, _ := url.ParseQuery(u.RawQuery)
+				path := u.Path
+				if strings.HasSuffix(path, ".js") {
+					query.Add("f", "sxg")

```

Comment:
I think we want to `Set` here, instead of `Add`. That'll allow us to overwrite the value (if there was one), or add it (if there wasn't).

Reply:
Done.